### PR TITLE
Redo gulp & webpack (WIP)

### PR DESCRIPTION
**Pull request by wolfd**
Adds:
- Browsersync
- Hot reloading JS and React
- Hot reloading css

Changes:
- Build pipeline now uses webpack for css

Still need to be done:
- Figure out if the production build actually works for our setup
- favicon.ico doesn't seem to work yet

### CODE REVIEW

File: SingularityUI/app/assets/index.mustache [hunk 1]
```diff
@@ -45,8 +51,14 @@
...
     </script>
+    {{#isDebug}}
+    <script src="/static/js/app.bundle.js"></script>
+    <script src="/static/js/vendor.bundle.js"></script>
+    {{/isDebug}}
+    {{^isDebug}}
+    <script src="{{{staticRoot}}}/js/app.bundle.js"></script>
     <script src="{{{staticRoot}}}/js/vendor.bundle.js"></script>
-    <script src="{{{staticRoot}}}/js/app.js"></script>
+    {{/isDebug}}
```
Comment (tpetr) [inline SingularityUI/app/assets/index.mustache]: Why not reduce complexity by forceing staticRoot to /static in gulp?

Comment (wolfd) [inline SingularityUI/app/assets/index.mustache]: Gulp drops the index.html file into the dist folder, but not in the static folder. I mostly copied the way it worked before, but I would be interested in changing things there if we are interested.

**Comment by Calvinp** [comment]
If this is still happening, please remember to add the case-sensitive paths plugin. ( #1202 )

**Comment by ssalinas** [comment]
@tpetr most gulp/webpack stuff got folded into the tailer PR yes? Can we close this guy or do we still need it?

**Comment by tpetr** [comment]
@ssalinas just double checked and all the features have been ported to the tailer branch. Closing this PR.

**Pull request closed without merging**